_posts/2022-01-19-7.0.0-release.md

```diff
@@ -0,0 +1,227 @@
+(ARROW-13494).
+
+Concatenation of union arrays is now supported (ARROW-4975).
+
+`StructType` gained three convenience methods to add, change and remove
+a given field (ARROW-11424).
+
+The `Datum` kind `COLLECTION` has been removed as it was entirely unused
+in the codebase (ARROW-13598).
+
+### Compute Layer
+
+A number of compute functions have been added:
+
+- functions operating on strings: "binary_reverse" (ARROW-14306),
+  "string_repeat" (ARROW-12712), "utf8_normalize" (ARROW-14205);
+- "fill_null_forward", "fill_null_backward" (ARROW-1699);
+- "ceil_temporal", "floor_temporal", "round_temporal" to adjust temporal input
+  to an integral multiple of a given unit (ARROW-14822);
+- "year_month_day" to extract the calendar components of the input (ARROW-15032);
+- "random" to general random floating-point values between 0 and 1 (ARROW-12404);
+- "indices_nonzero" to return the indices in the input where there are
+  non-zero, non-null values (ARROW-13035).
+
+Decimal data is now supported as input of the arithmetic kernels
+(ARROW-13130).
+
+Dictionary data is now supported as input of the hash join execution node
+(ARROW-14181).
+
+Residual predicates have been implemented in the hash join node
+(ARROW-13643).
+
+The "list_parent_indices" function now always returns int64 data
+regardless of the input type (ARROW-14592).
+
+Month-day-nano interval data is now supported as input of the same functions
+as other interval types (ARROW-13989).
+
+### CSV
+
+The CSV writer got additional configuration options:
+- the string representation of null values (ARROW-14905);
+- the quoting strategy: always / never / as needed (ARROW-14905);
+- the end of line character(s) (ARROW-14907)
+
+### Dataset Layer
+
+"SkyHook", a dataset addition that offloads fragment scan operations to a
+Ceph distributed storage, was contributed (ARROW-13607).
+
+The dataset writer now exposes options `min_rows_per_group` and
+`max_rows_per_group` to control the size of row groups created (ARROW-14426).
+
+### IO and Filesystem Layer
+
+A critical bug in the AWS SDK for C++ that risks losing data in S3 multipart
+uploads has been circumvented (ARROW-14523).
+
+The Google Cloud Storage filesystem is now featureful enough to pass all
+generic filesystem tests (ARROW-14924).
+
+The OpenAppendStream method of filesystems has been un-deprecated; however,
+it still cannot be implemented for all filesystem backends (ARROW-14969).
+
+A new function `arrow::fs::ResolveS3BucketRegion` allows resolving the
+region where a particular S3 bucket resides (ARROW-15165).
+
+The S3 filesystem now sets the Content-Type of output files to
+"application/octet-stream" (instead of "application/xml" previously)
+if not explicitly specified by the caller (ARROW-15306).
+
+### IPC
+
+Fine-grained I/O (coalescing) is now enabled in the synchronous (ARROW-12683)
+and asynchronous (ARROW-14577) IPC reader.
+
+It is now possible to set the compression level when using LZ4 compression
+(ARROW-9648).
+
```

Comment:
### ORC

The ORC adapters have been significantly improved. A lot more properties of the ORC reader as well as ORC writer options are now available. Moreover API docs for both the ORC reader and the ORC writer have been generated.  (ARROW-11297)